syntax = "proto3";
package workload;




import "container_sdk/model/container/event.proto";

/*
ListEvent请求
*/
message ListEventRequest {
    /*
    workloadId
    */
    string instanceId = 1 ;
    /*
    类型，Pod/ReplicaSet
    */
    string kind = 2 ;
    /*
    事件级别
    */
    string type = 3 ;
}

/*
ListEvent返回
*/
message ListEventResponse {
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated container.Event list = 4 ;
}

/*
ListEventApi返回
*/
message ListEventResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    workload.ListEventResponse data = 4 ;
}

